src/core/utils/assembly.ts

```diff
@@ -18,19 +18,48 @@ export interface ChromSize {
 
 /**
  * Get relative chromosome position (e.g., `100` => `{ chromosome: 'chr1', position: 100 }`)
+ * @param absPos number which is the absolute chromosome position
+ * @param assembly the assembly used to calculate which chromosome position
+ * @param returnWithinAssembly If true, then if the absolute position is before the first chromosome, it returns the
+ * first position of the first chromosome. If the absolute position is after the last chromosome, it returns the last
+ * position of the last chromosome
+ * @returns the genomic position of the absPos
  */
-export function getRelativeGenomicPosition(absPos: number, assembly?: Assembly): GenomicPosition {
-    const [chromosome, absInterval] = Object.entries(computeChromSizes(assembly).interval).find(d => {
-        const [start, end] = d[1];
-        return start <= absPos && absPos < end;
-    }) ?? [null, null];
-
-    if (!chromosome || !absInterval) {
-        // The number is out of range
+export function getRelativeGenomicPosition(
+    absPos: number,
+    assembly?: Assembly,
+    returnWithinAssembly = false
```

Comment:
Great. Do you mind adding a test for this option?

Reply:
Added!